tests/integration/test_parallel_replicas_working_set/test.py

```diff
@@ -0,0 +1,140 @@
+import pytest
+from helpers.cluster import ClickHouseCluster
+
+cluster = ClickHouseCluster(__file__)
+
+nodes = [
+    cluster.add_instance(
+        f"n{i}", main_configs=["configs/remote_servers.xml"], with_zookeeper=True
+    )
+    for i in (1, 2, 3)
+]
+
+
+@pytest.fixture(scope="module", autouse=True)
+def start_cluster():
+    try:
+        cluster.start()
+        yield cluster
+    finally:
+        cluster.shutdown()
+
+
+def create_tables(cluster, table_name, node_with_covering_part):
+    # create replicated tables
+    for node in nodes:
+        node.query(f"DROP TABLE IF EXISTS {table_name} SYNC")
+
+    nodes[0].query(
+        f"""CREATE TABLE IF NOT EXISTS {table_name} (key Int64, value String) Engine=ReplicatedMergeTree('/test_parallel_replicas/shard1/{table_name}', 'r1')
+            ORDER BY (key)"""
+    )
+    nodes[1].query(
+        f"""CREATE TABLE IF NOT EXISTS {table_name} (key Int64, value String) Engine=ReplicatedMergeTree('/test_parallel_replicas/shard1/{table_name}', 'r2')
+            ORDER BY (key)"""
+    )
+    nodes[2].query(
+        f"""CREATE TABLE IF NOT EXISTS {table_name} (key Int64, value String) Engine=ReplicatedMergeTree('/test_parallel_replicas/shard1/{table_name}', 'r3')
+            ORDER BY (key)"""
+    )
+    # stop merges to keep original parts
+    # stop fetches to keep only parts created on the nodes
+    for i in (0, 1, 2):
+        if i != node_with_covering_part:
+            nodes[i].query(f"system stop fetches {table_name}")
+            nodes[i].query(f"system stop merges {table_name}")
+
+    # populate data, equal number of rows for each replica
+    nodes[0].query(
+        f"INSERT INTO {table_name} SELECT number, number FROM numbers(10)",
+    )
+    nodes[0].query(
+        f"INSERT INTO {table_name} SELECT number, number FROM numbers(10, 10)"
+    )
+    nodes[1].query(
+        f"INSERT INTO {table_name} SELECT number, number FROM numbers(20, 10)"
+    )
+    nodes[1].query(
+        f"INSERT INTO {table_name} SELECT number, number FROM numbers(30, 10)"
+    )
+    nodes[2].query(
+        f"INSERT INTO {table_name} SELECT number, number FROM numbers(40, 10)"
+    )
+    nodes[2].query(
+        f"INSERT INTO {table_name} SELECT number, number FROM numbers(50, 10)"
+    )
+    nodes[node_with_covering_part].query(f"system sync replica {table_name}")
+    nodes[node_with_covering_part].query(f"optimize table {table_name}")
+
+    # check we have expected set of parts
+    expected_active_parts = ""
+    if node_with_covering_part == 0:
+        expected_active_parts = (
+            "all_0_5_1\nall_2_2_0\nall_3_3_0\nall_4_4_0\nall_5_5_0\n"
```

Comment:
This will fail with ZK retries (different block number), right? You might want to disable fault injections if that's the case

Reply:
Thanks, but are we gonna to enable zk fault injections in integration test?

Reply:
True. I thought they were already on by default but they aren't.